### PR TITLE
Clear hostname from /etc/hosts when deleting nic config

### PR DESCRIPTION
**Pull request by mchf**
Ported patch https://github.com/yast/yast-network/pull/1
Ported patch for bnc#951330

### CODE REVIEW

File: src/modules/Host.rb [hunk 1]
```diff
@@ -57,12 +57,39 @@ def main
...
+
+    # Give address a new list of names.
+    def set_names(address, names)
+      @hosts[address] = names
+      modified = true
+    end
+
+    # Add another name to the list for address (which may be empty so far)
+    def add_name(address, name)
+      @hosts[address] = [] if @hosts[address].nil?
```
Comment (jreidinger) [inline src/modules/Host.rb]: `@hosts[address] ||= []`

File: src/modules/Host.rb [hunk 2]
```diff
@@ -349,9 +376,7 @@ def SetModified
       nil
     end
 
-    publish :variable => :hosts, :type => "map <string, list <string>>"
     publish :variable => :modified, :type => "boolean"
-    publish :variable => :write_only, :type => "boolean"
```
Comment (jreidinger) [inline src/modules/Host.rb]: this removes also accessor, do you really check all SLE12 modules? It is API incompatible change, which is not nice in update branch

File: src/modules/Host.rb [hunk 3]
```diff
@@ -57,12 +57,39 @@ def main
...
-      # Only write configuration
-      @write_only = false
-
       @initialized = false
     end
 
+    # Remove all entries from the host table.
+    def clear
+      @hosts = {}
+      modified = true
```
Comment (mvidner) [inline src/modules/Host.rb]: If we replace all `modified = true` with `@modified = true` then we can remove the only remaining`publish :variable`.
It is that way in the original patch. Thanks for porting it, it looks like I missed to merge it back then :-/

File: src/modules/Host.rb (hunk 2 above)
Comment (mvidner) [inline src/modules/Host.rb]: I have double checked that these variables are indeed not used in other packages.

File: src/modules/Host.rb (hunk 3 above)
Comment (jreidinger) [inline src/modules/Host.rb]: btw this is wrong. it have to be `self.modified = true` otherwise it is assigned to local variable and lost.

Comment (mchf) [inline src/modules/Host.rb]: I've obviously missed the modification mentioned by @mvidner above. I'll update the PR

**Comment by mvidner** [comment]
LGTM

**Comment by mchf** [comment]
Thanks for reviews

**Pull request merged**